### PR TITLE
[docs] node affinity docs; 0.2.0 release prep

### PR DESCRIPTION
**Pull request by schallert**
- Changelog entries for 0.2.0
- Docs on new features (tolerations, expanded node affinity)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,42 @@
...
+Features such as specifying etcd endpoints directly on the cluster spec eliminate the need to provide a manual
+configuration for custom etcd endpoints. Per-cluster etcd environments will allow users to collocate multiple m3db
+clusters on a single etcd cluster.
+
+Users can now specify more complex affinity terms, and specify taints that their cluster tolerates to allow dedicating
+specific nodes to M3DB. See the [affinity docs][affinity-docs] for more.
+
+* [FEATURE] Allow specifying of etcd endpoints on M3DBCluster spec ([#99][99])
+* [FEATURE] Allow specifying security contexts for M3DB pods ([#107][107])
+* [FEATURE] Allow specifying tolerations of m3db pods ([#111][111])
```
Comment (robskillington) [inline CHANGELOG.md]: nit: `m3db` -> `M3DB` for consistency

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,13 +52,28 @@
 
 ## 0.1.0
 
-* TODO
+* Fix helm manifests.
 
 ## 0.1.0
```
Comment (robskillington) [inline CHANGELOG.md]: Seems like there are two entries for 0.1.0 here - maybe just combine?

File: CHANGELOG.md [hunk 3]
```diff
@@ -18,13 +52,28 @@
...
 
-* TODO
+* Fix helm manifests.
```
Comment (robskillington) [inline CHANGELOG.md]: Should the header for this be `0.1.1`?

File: CHANGELOG.md [hunk 4]
```diff
@@ -18,13 +52,28 @@
...
 [100]: https://github.com/m3db/m3db-operator/pull/100
+[106]: https://github.com/m3db/m3db-operator/pull/106
+[107]: https://github.com/m3db/m3db-operator/pull/107
+[111]: https://github.com/m3db/m3db-operator/pull/111
+[112]: https://github.com/m3db/m3db-operator/pull/112
+[114]: https://github.com/m3db/m3db-operator/pull/114
+[117]: https://github.com/m3db/m3db-operator/pull/117
+[118]: https://github.com/m3db/m3db-operator/pull/118
+[119]: https://github.com/m3db/m3db-operator/pull/119
+[99]: https://github.com/m3db/m3db-operator/pull/99
```
Comment (robskillington) [inline CHANGELOG.md]: nit: Worth ordering 99 above 100 here?

File: docs/configuration/node_affinity.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+Kubernetes allows pods to be assigned to nodes based on various critera through [node affinity][k8s-node-affinity].
+
+M3DB was built with failure tolerance as a core feature. M3DB's [isolation groups][m3db-isogroups] allow shards to be
+placed across failure domains such that the loss of no single domain can cause the cluster to lose quorum. More details
+on M3DB's resiliency can be found in the [deployment docs][m3db-deployment].
+
+By leveraging Kubernetes' node affinity and M3DB's isolation groups, the operator can guarantee that M3DB pods are
+distributed across failure domains. For example, in a Kubernetes cluster spread across 3 zones in a cloud region, the
+`isolationGroups` below config would guarantee that no single zone failure could degrade the M3DB cluster.
```
Comment (robskillington) [inline docs/configuration/node_affinity.md]: nit: `below config` -> `configuration below`

File: docs/configuration/node_affinity.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+    numInstances: 3
+    nodeAffinityTerms:
+    - key: failure-domain.beta.kubernetes.io/zone
+      values:
+      - us-east1-d
+    - key: nodepool
+      values:
+      - m3db-dedicated-pool
+  tolerations:
+  - key: m3db-dedicated
```
Comment (robskillington) [inline docs/configuration/node_affinity.md]: Should this need to be `m3db-dedicated-pool` or is `m3db-dedicated` and `m3db-dedicated-pool` somehow related somewhere?

File: scripts/prep_changelog.sh [hunk 7]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# Script for transforming non-hyperlink changelog entries into ones with
+# hyperlinks using an opionated toolset (https://github.com/chmln/sd and
+# https://github.com/BurntSushi/ripgrep).
```
Comment (robskillington) [inline scripts/prep_changelog.sh]: Probably not worth doing now, but is there a way to bundle these in some cargo deps list so users can just `make dependencies` and get these installed for free locally?

File: docs/configuration/node_affinity.md (hunk 6 above)
Comment (schallert) [reply docs/configuration/node_affinity.md]: So `m3db-dedicated-pool` the nodepool and `m3db-dedicated` the taint are two separate things, I wanted to convey that they could be named differently (like `foo` and `bar` even). Maybe I should make them even more different to clarify that? Or can just mention it

File: scripts/prep_changelog.sh (hunk 7 above)
Comment (schallert) [reply scripts/prep_changelog.sh]: I should just stop being lazy and make it work with `grep` and `sed`

**Comment by schallert** [comment]
Landing now to get fixes on a tagged release, will follow up for doc feedback if there's more.

**Pull request merged**